### DISCUSSION TITLE
Easy to use/install binaries for Windows _including_ the library

### DISCUSSION THREAD

**Discussion by mmomtchev** [ImageMagick/ImageMagick#6293] [Development] [answered]
### ImageMagick version

7.1.1

### Operating system, version and so on

Windows

### Description

There is a separate discussion about replacing the aging ImageMagick build system - however replacing the build system of a complex and highly portable C++ project that goes back 30 years won't happen overnight.
The current build system has only one major problem and this is automated builds on Windows. Currently, if one wants to distribute an open-source project that uses ImageMagick, there are simply no viable options on Windows.

Would it be possible to include the library and the header files in the prebuilt binaries? Automated builds have always been a weak spot on Windows, but on this platform, one binary serves all. This is a much simpler alternative to completely replacing the build system - which is the long-term solution.

A ZIP-file, on a well-known URL, that contains the `.h`  and the `.lib` files.

**Comment by dlemstra** [comment, marked as answer]
You can automate the build of ImageMagick and we are doing that to create the binaries for the installer on Windows. We will not create an extra zip file that contains these files for you. What you could do is use the installer of ImageMagick to install the library and select that option to install the Development libraries and copy the `.h` and `.lib` files from the installation folder and place them somewhere where you can use them inside your project.

**Reply by mmomtchev** [reply]
Ok, I didn't know that the library/headers were available in the installer. But, still, those are not really usable inside of a Github Action. At the moment I am using your sample GA workflow which is probably the best choice.

**Reply by dlemstra** [reply]
There might be an option to use the installer on the command line and use that to install the library from the command line. And if that is not possible feel free to open a PR to add support for this. Or maybe tou can use this: https://community.chocolatey.org/packages/imagemagick.app?
